distr/flecs.h

```diff
@@ -15331,6 +15331,33 @@ typedef struct ecs_serializer_t {
 extern "C" {
 #endif
 
+/** Visitor Seriailzation API
+ */
+typedef struct ecs_visitor_desc_t {
+    /** Here Be Visitor Functions
+     * If the visit has a return, it has some form of flow control.
+     */
+
+    /** Visit Primitives */
+    void (*visit_f32)(const ecs_f32_t value, void* user_data);
+    void (*visit_f64)(const ecs_f64_t value, void* user_data);
+
```

Comment:
This not appear to have all the same functions that appear in `include/flecs/addons/meta.h`?

Reply:
Those are functions for opaque type serialization. Currently all that is uploaded is an initial (wip) commit since I'm still working on a commit after raising the abstraction all the way to `ecs_array_to_json` and similar serialization entry points. 

To get the gist, you can see https://github.com/SanderMertens/flecs/pull/1456/files#diff-183bddbbb7fb95a645f2bf0b20f055456269a7d4994f37e63a60844b4042ca12R155. The struct you quoted is passed into the serialization functions for each type you want to use. It will call into those opaque serialization functions you mention.